numba/tests/test_linalg.py

```diff
@@ -805,25 +817,31 @@ def sample_matrix(self, m, dtype, order):
         Q = np.array(Q, dtype=dtype, order=order)
         return Q
 
-    def assert_no_domain_change(self, cfunc, args):
-        msg = "eig() argument must not cause a domain change."
+    def assert_no_domain_change(self, name, cfunc, args):
+        msg = name + "() argument must not cause a domain change."
         self.assert_error(cfunc, args, msg)
 
-    @needs_lapack
-    def test_linalg_eig(self):
+    def checker_for_linalg_eig(
+            self, name, func, expected_res_len, check_for_domain_change=None):
         """
         Test np.linalg.eig
         """
         n = 10
-        cfunc = jit(nopython=True)(eig_matrix)
+        cfunc = jit(nopython=True)(func)
 
         def check(a):
-            expected = eig_matrix(a)
+            expected = func(a)
             got = cfunc(a)
             # check that the returned tuple is same length
             self.assertEqual(len(expected), len(got))
-            # and that length is 2
-            self.assertEqual(len(got), 2)
+            # and that dimension is correct
+            res_is_tuple = False
+            if isinstance(got, tuple):
+                res_is_tuple = True
+                self.assertEqual(len(got), expected_res_len)
```

Comment:
It's comparing to a hardcoded known value that is supplied in the arguments to the checker function. There is a comparison to `expected` itself here:
https://github.com/numba/numba/pull/2129/files#diff-3208831c404c0a8b312ff145797b3818R836
I can change/remove this if you wish?